isis/tests/FunctionalTestsCamstats.cpp

```diff
@@ -0,0 +1,154 @@
+
+  group = appLog.findGroup("AspectRatio");
+  EXPECT_DOUBLE_EQ( (double) group.findKeyword("AspectRatioMinimum"), 1.0);
+  // Maximum spelled incorrectly to match misspelling in CameraStatistics.cpp
+  EXPECT_DOUBLE_EQ( (double) group.findKeyword("AspectRatioMaximun"), 1.0);
+  EXPECT_DOUBLE_EQ( (double) group.findKeyword("AspectRatioAverage"), 1.0);
+  EXPECT_DOUBLE_EQ( (double) group.findKeyword("AspectRatioStandardDeviation"), 0.0);
+
+  group = appLog.findGroup("PhaseAngle");
+  EXPECT_NEAR( (double) group.findKeyword("PhaseMinimum"), 79.756143590, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("PhaseMaximum"), 81.304900313, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("PhaseAverage"), 80.529097153, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("PhaseStandardDeviation"), 0.444208612, 1e-8);
+
+  group = appLog.findGroup("EmissionAngle");
+  EXPECT_NEAR( (double) group.findKeyword("EmissionMinimum"), 10.798462835, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("EmissionMaximum"), 13.502630463, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("EmissionAverage"), 12.15148695101, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("EmissionStandardDeviation"), 0.565437913, 1e-8);
+
+  group = appLog.findGroup("IncidenceAngle");
+  EXPECT_NEAR( (double) group.findKeyword("IncidenceMinimum"), 69.941096124, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("IncidenceMaximum"), 70.311944975, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("IncidenceAverage"), 70.127459134, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("IncidenceStandardDeviation"), 0.102490391, 1e-8);
+
+  group = appLog.findGroup("LocalSolarTime");
+  EXPECT_NEAR( (double) group.findKeyword("LocalSolarTimeMinimum"), 7.7698055422, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("LocalSolarTimeMaximum"), 7.8031735959, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("LocalSolarTimeAverage"), 7.7863626216, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("LocalSolarTimeStandardDeviation"), 0.007105554, 1e-8);
+
+  group = appLog.findGroup("LocalRadius");
+  EXPECT_NEAR( (double) group.findKeyword("LocalRadiusMinimum"), 3410663.3374636001, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("LocalRadiusMaximum"), 3413492.0662691998, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("LocalRadiusAverage"), 3412205.8144924999, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("LocalRadiusStandardDeviation"), 648.5763070953, 1e-8);
+
+  group = appLog.findGroup("NorthAzimuth");
+  EXPECT_NEAR( (double) group.findKeyword("NorthAzimuthMinimum"), 312.299406589, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("NorthAzimuthMaximum"), 350.597812506, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("NorthAzimuthAverage"), 332.967661510, 1e-8);
+  EXPECT_NEAR( (double) group.findKeyword("NorthAzimuthStandardDeviation"), 0.673831891, 1e-8);
+}
+
+TEST_F(DefaultCube, FunctionalTestCamstatsAttach) {
+  QVector<QString> args = {"attach=true", "linc=100", "sinc=100"};
```

Comment:
Set the linc and sinc parameters to speed up the test runtime. This cuts it down from ~56 seconds to ~0.6 seconds. Is this worth doing for the DefaultParameters test case as well, or is it better to leave one in that hits all lines/samples?

Reply:
I don't think we need to do every pixel. We just want to test that adjusting it changes the output.